File: Programmers/Lv.2/다음 큰 숫자/12911.java
class Solution {
    public int solution(int n) {
        int answer = n;
        
        String binaryN = Integer.toBinaryString(n);
        int countN = 0;
        for (char c : binaryN.toCharArray()) {
            if (c == '1') countN++;
        }
        
        int count = 0;
        while (count != countN) {
            count = 0;
            
            String binary = Integer.toBinaryString(++answer);
            for (char c : binary.toCharArray()) {
                if (c == '1') count++;
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.2/다음 큰 숫자/12911.js
function solution(n) {
    let answer = n;
    
    const countN = [...n.toString(2)].filter(v => v === "1").length;
    
    let count = 0;
    while (count !== countN) {
        count = [...(++answer).toString(2)].filter(v => v === "1").length;
    }
    
    return answer;
}
